### PR TITLE
Update @wordpress/scripts dependency in blocks

### PR DESCRIPTION
**Pull request by samueljseay**
#### Update `@wordpress/scripts`
Update the `@wordpress/scripts` dependency used in the `@woocommerce/block-library` package to the latest version.

#### Fix Jest after dependency changes
When updating this package we get a few newer dependencies which break our Jest config, here's a rough list of changes:

1. Remove transform ignore paths we don't need anymore now that a lot of these libs are better handled by Jest 28+
2. For a couple of problematic packages we include some fixes:
  a. parsel-js has some errors in how export priority is declared, so it tries to resolve ESM. use custom resolver to find right file
  b. uuid v8 specifically also tries to resolve ESM, we use custom resolver to tell it not to
  c. client-zip does not provide a non-esm module, we just mock it out because we do not need to test this lib directly
3. a newer version of `@wordpress/viewport` uses `window.matchMedia` which Jest DOM does not support. We add a mock for this.
4. Snapshot's have been updated. I checked each one and it's not a change in how our UI is rendered and they still all look correct, but rather it's a subtle change in the snapshot format. For example, snapshots use to have syntax like `Object { ...` but now are generated as `{ ...`
  
FWIW the custom resolver solution is much nicer than trying to transpile some modules since the transpiled code is indeed available, it's just that Jest tries to use the wrong one based on how the package json is written.

### How to test the changes in this Pull Request:

It will be suffice to see the CI pass but please also have a look carefully over the snapshots as I did to ensure nothing looks awry.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Update the `@wordpress/scripts` dependency used in the `@woocommerce/block-library` package to 30.6.0 
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Review by louwie17 — APPROVED** [review]
Thanks for working on this @samueljseay, and figuring out the intricacies with Jest especially around the ESM stuff.
The changes look good and I also ran the tests locally and it all ran successfully. I also started looking if the blocks themselves still work, but then realized we don't use `@wordpress/scripts` for our build process in the blocks package.

Would be nice to have @joshuatf still review, but LGTM if you ask me :)

File: plugins/woocommerce-blocks/tests/js/jest.config.json
Comment (joshuatf) [inline plugins/woocommerce-blocks/tests/js/jest.config.json:55]: Thanks for removing these 🙏

File: plugins/woocommerce-blocks/tests/js/mocks/client-zip.js
Comment (joshuatf) [inline plugins/woocommerce-blocks/tests/js/mocks/client-zip.js:1]: Good call 👍

**Review by joshuatf — APPROVED** [review]
Looks good and all tests are passing!  Appreciate the clean-up around the old transforms as well.  :shipit:

**Pull request merged**
